site/src/pages/dashboard/[[...slugs]].page.tsx

```diff
@@ -9,81 +8,35 @@ import {
-} from "../../components/pages/dashboard/dashboard-card";
+import { DashboardCard } from "../../components/pages/dashboard/dashboard-card/dashboard-card";
+import { SectionGrid } from "../../components/pages/dashboard/dashboard-section-grid";
+import { SectionTitle } from "../../components/pages/dashboard/dashboard-section-title";
+import { PageContainer } from "../../components/pages/dashboard/page-container";
 import { TopNavigationTabs } from "../../components/pages/dashboard/top-navigation-tabs";
-
-const getDashboardPrimaryCardData = (
-  openCreateSchemaModal: () => void,
-): DashboardCardProps[] => [
-  {
-    title: "Read our quick start guide to building blocks",
-    colorGradient:
-      "linear-gradient(291.34deg, #FFB172 -4.12%, #D082DE 53.49%, #9482FF 91.74%, #84E6FF 151.68%)",
-    description:
-      "Learn how to build blocks that can integrate with the block protocol and be published to the registry.",
-    link: {
-      title: "Read the guide",
-      href: "/docs/developing-blocks",
-    },
-  },
-  {
-    title: "Test out blocks in the playground",
-    colorGradient:
-      "linear-gradient(310.17deg, #FFB172 -167.67%, #9482FF 13.54%, #84E6FF 126.83%)",
-    description:
-      "Try out our most popular blocks to learn how they consume and render different data structures",
-    link: {
-      title: "Try out blocks",
-      href: "/hub",
-    },
-  },
-  {
-    title: "Create a Schema",
-    colorGradient:
-      "linear-gradient(91.21deg, #FFB172 -84.62%, #9482FF 62.56%, #84E6FF 154.58%)",
-    description:
-      "Schemas are a formal way to describe data – they define the expected properties of things like People, Companies, and Books",
-    link: {
-      title: "Create a schema",
-      onClick: openCreateSchemaModal,
-    },
-  },
-];
-
-const getDashboardSecondaryCardData = (
-  profileLink: string,
-): DashboardCardProps[] => [
-  {
-    title: "Generate your API key",
-    description:
-      "Your API key will allow you to search for blocks by name, author, or compatible data structure.",
-    link: {
-      title: "Generate Key",
-      href: "/settings/api-keys",
-    },
-    icon: faKey,
-  },
-  {
-    title: "View your public profile",
-    description:
-      "See what others see and browse your published blocks and schemas",
-    link: {
-      title: "View Profile",
-      href: profileLink,
-    },
-    icon: faUser,
-    variant: "secondary",
-  },
-];
+import {
+  DashboardSection,
+  getDashboardSectionCards,
+} from "../../components/pages/dashboard/utils";
+import { apiClient } from "../../lib/api-client";
 
 const Dashboard: AuthWallPageContent = ({ user }) => {
   const { preferredName: userName, shortname } = user ?? {};
   const [schemaModalOpen, setSchemaModalOpen] = useState(false);
+  const [userHasApiKey, setUserHasApiKey] = useState<boolean | undefined>(
+    undefined,
+  );
   const router = useRouter();
 
+  useEffect(() => {
+    void apiClient
+      .getUserApiKeys()
```

Comment:
If getUserApiKeys fails for some reason, the page will keep loading. We should handle setting userHasApiKey in that scenario

Reply:
I think it might make more sense to add a `hasApiKey` to the user object serverside, so we don't have to make this call in the client. We can do this in the serialize method here => https://github.com/blockprotocol/blockprotocol/blob/main/site/src/lib/api/model/user.model.ts#L447-L455

Reply:
It will also be great if we can then have the skeletons showing up the first time we fetch the current user

Reply:
Fixed possible infinite loading issue here https://github.com/blockprotocol/blockprotocol/pull/511/commits/546ba3dcd6f8dbdae1f2ae8045e0ab5eeb4f1896.

@teenoh About adding loading state to the whole page depending on user or other params, I feel like it's out of scope of this PR. I think we should develop a re-useable solution to showing loading states based on params at some point. But for now we only have 1 page has skeleton loading state, which is dashboard. And the current loading is just a glimpse. So I feel like we can postpone having a better loading state solution when we start rendering more user related data on the dashboard.

Reply:
Totally understand. In my comment, I was actually referring to having the skeletons in the dashboard page when the user is being fetched (user === 'loading'), but I agree we need to come up with a more reusable solution to show loading states on other pages and that's outside the scope of this PR.
 
In the mean time, I think it should be a bit straightforward to show the skeletons when the user is being fetched in dashboard page.

Currently, reloading the dashboard page looks like this => 

https://user-images.githubusercontent.com/16928688/185929274-4ab49aef-9c8b-4f1c-934c-16106e81a09d.mov

If this takes longer than expected, then we can address in a follow up

Reply:
@CiaranMn what do you think about including `hasApiKey` in the user object?

Reply:
My instinct is that it's a bit wasteful to check for a user's API keys whenever we want to retrieve the user, when we really only need it for a couple of pages here – if the aim is to have the data available on the first render, can we make the request to `getUserApiKeys` in [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props) instead?

Reply:
@CiaranMn For that, we need to get the username on serverside. I checked how our authentication flow is written quickly, seems like we only have `blockprotocol-session-id` on serverside. And couldn't see any examples of it's being used in serverside. So I'm not sure how to implement it on `getServerSideProps`, or is it supported by our authentication flow right now.

Reply:
Ah, okay. We would need to move the call to get the user to server side then. Possibly we could use something like `getSession` in `session.middleware.ts`. There's a task ([internal link](https://app.asana.com/0/1200211978612931/1201715945634534/f)) for moving the user call to server side.

In the meantime, we could just not vary the API key text based on whether has an API key or not. We don't even allow users to have one more than one key at the moment so it's inaccurate as it is - you can either (a) create or (b) regenerate an API key. @nonparibus preference?

Reply:
Fine to not vary it for now. Eventually we will definitely want users to be able to have multiple keys but there is no immediate urgency re: this.